Pear/Controller/ChatViewController.swift

```diff
@@ -0,0 +1,331 @@
+
+        let rightPaddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 0))
+        chatInputTextField.rightView = rightPaddingView
+        chatInputTextField.rightViewMode = .always
+
+        sendMessageButton.setImage(UIImage(named: "sendMessage"), for: .normal)
+        sendMessageButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
+        chatInputContainerView.addSubview(sendMessageButton)
+
+        separatorView.backgroundColor = .inactiveGreen
+        chatInputContainerView.addSubview(separatorView)
+    }
+
+    private func getChatMessages() {
+        databaseRef.child("User-Messages").child(currentUser.netID).child(messageUser.netID).observe(.childAdded) { snapshot in
+            let messageId = snapshot.key
+            let messageRef = self.databaseRef.child("Messages").child(messageId)
+            messageRef.observeSingleEvent(of: .value) { snapshot in
+                self.getMessages(snapshot: snapshot)
+            }
+        }
+    }
+
+    private func getMessages(snapshot: DataSnapshot) {
+        let message = PearMessage(snapshot: snapshot)
+        self.messages.append(message)
+        DispatchQueue.main.async {
+            self.groupMessagesByDate()
+        }
+    }
+
+    private func groupMessagesByDate() {
+        self.groupedMessagesByDate = []
+        let groupedMessages = Dictionary(grouping: messages) { message -> Date in
+            var day = NSDate()
+            if let timeSince1970 = Double(message.time) {
+                day = NSDate(timeIntervalSince1970: timeSince1970)
+            }
+            var date = day as Date
+            date = stripTimeOffDate(from: date)
+            return date
+        }
+
+        let sortedKeys = groupedMessages.keys.sorted()
+        sortedKeys.forEach { key in
+            let values = groupedMessages[key]
+            self.groupedMessagesByDate.append(values ?? [])
+        }
+
+        dateKeys = sortedKeys
+
+        guard var bottomRow = self.groupedMessagesByDate.last?.count else { return }
+        bottomRow = bottomRow - 1
+        DispatchQueue.main.async {
+            self.chatTableView.reloadData()
+            let bottomSection = self.groupedMessagesByDate.count - 1
+            self.scrollToRow(row: bottomRow, section: bottomSection)
+        }
+    }
+
+    private func scrollToRow(row: Int, section: Int) {
+        DispatchQueue.main.async {
+            self.chatTableView.reloadData()
+            let indexPath = IndexPath(row: row, section: section)
+            if self.chatTableView.validIndexPath(indexPath: indexPath) {
+                self.chatTableView.scrollToRow(at: indexPath, at: .bottom, animated: true)
+            }
+        }
+    }
+
+    private func stripTimeOffDate(from originalDate: Date) -> Date {
+        let components = Calendar.current.dateComponents([.year, .month, .day], from: originalDate)
+        let date = Calendar.current.date(from: components)
+        return date!
```

Comment:
also might be better to return `date ?? originalDate` than crash from force unwrap